### PR TITLE
fix: update semantic scores based on dataclass fields of query

### PR DESCRIPTION
**Pull request by violenil**
Previously, when defining semantic scores, one could not define query fields with arbitrary names. The API only accepted semantic scores defined with query fields `'text_0', 'text_1' ...` and `'image_0', 'image_1' ...`. However, the query field in the semantic scores should be allowed to be any string name, as long as it is **consistent** with the name defined in the `'name'` of the query. For example, one would expect that the following snippet works with query field name `'my_query_text'`:

```
import requests

HOST = "<my_host_url>"

request_body = {
    'query': [{'name':'my_query_text', 'modality': 'text', 'value': 'test'}],
    'semantic_scores': [('my_query_text', 'label', 'encoderclip', 1.0)]
 }

response = requests.post('{HOST}/api/v1/search-app/search', json=request_body)
print(response.status_code)
print(response.content)
```

This PR fixes this issue and allows arbitrary query field naming, by mapping this query field name back to the dataclass field that is created for the query document.

Successful customer deployment:
Screenshot 2023-02-24 at 15 09 49

---

- [x] This PR references an open issue
- [x] Added a test
- [ ] Updated the documentation
- [x] Added a screenshot of a successful customer deployment

### CODE REVIEW

**Review by joschkabraun — APPROVED** [review]
LGTM

File: tests/integration/offline_flow/test_offline_flow.py [hunk 1]
```diff
@@ -34,6 +34,7 @@ async def test_docarray(
     search_result = await search(
         SearchRequestModel(
             query=[{'name': 'text', 'value': 'girl on motorbike', 'modality': 'text'}],
+            semantic_scores=[('text', 'product_title', 'encoderclip', 1.0)],
```
Comment (Tanguyabel) [inline tests/integration/offline_flow/test_offline_flow.py:37]: Wouldn't it make sense to create another test with semantic scores, rather than updating this one?

Comment (violenil) [reply tests/integration/offline_flow/test_offline_flow.py:37]: There is a unit test for the bff for testing semantic scores [here](https://github.com/jina-ai/now/blob/main/tests/unit/bff/v1/routers/test_search.py#L110) but it did not fail on this edge case, probably because the problem is related to the indexer not being able to handle the semantic scores.

What might help is to add the semantic scores to other integration tests that run full deployments. I can make this change.

Comment (Tanguyabel) [reply tests/integration/offline_flow/test_offline_flow.py:37]: Sounds good to me. In addition, we should ensure that `client_with_mocked_jina_client` in `bff/conftest.py` can support semantic scores as well.

**Review by Tanguyabel — APPROVED** [review]
looks good to me!

**Comment by violenil** [comment]
After changes in bff and playground, customer deployment still working:
Screenshot 2023-03-01 at 23 59 14

**Review by makram93 — APPROVED** [review]
